### DISCUSSION TITLE
When you catch Copilot reading your code!

### DISCUSSION THREAD

**Discussion by dolanbs1959** [community/community#111006] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I'm curious (and a bit creeped out) about this topic. I haven't been using AI long, and I'm still in my trial period with GitHub Copilot. But I have been using it daily for debugging and working through errors I've had. It's been extremely helpful, and AI in general is proving to be a powerful resource. I understood that this is all it is too, just a resource, or a tool to get a job done. That's what I thought, until yesterday. 
I had prompted a simple question about spacing between elements and it quickly responded with an answer, but included actual data I am using in my code. We're talking specific text I am using in labels, none of which I have shared prior to this prompt. These labels are not common "default" labels by any means, as they are specific to my project criteria. Nonetheless, it read my code and provide a real-time example. It makes me wonder how Skynet started out?
Of course, asking how it was able to return my specific text, it said it was a coincidence and not based on any specific knowledge of my application. 

> "I don't have the ability to access or retrieve personal data unless it has been shared with me in the course of our conversation. I am designed to respect user privacy and confidentiality. The options I provided were purely illustrative and any match with your actual data is coincidental."

Obviously it's going to plead the 5th! Lol!!

I was curious if anyone else has noticed this odd and somewhat concerning behavior.

**Comment by Akash1134** [comment]
Hi @dolanbs1959 ,

This particular documentation answers your question: [About privacy for GitHub Copilot Individual](https://docs.github.com/en/enterprise-cloud@latest/copilot/copilot-individual/about-github-copilot-individual#about-privacy-for-github-copilot-individual)

While the above response you got from Copilot is in context of both ensuring your privacy and certain limitations of the tool. You can read more about it [here](https://docs.github.com/en/copilot/github-copilot-chat/about-github-copilot-chat#limitations-of-github-copilot-chat).

Thanks!
